Split/Storage/Impressions/ImpressionsMode.swift

```diff
@@ -24,3 +24,39 @@ enum ImpressionsMode: String {
         }
     }
 }
+
+@propertyWrapper
+public struct ImpressionsModeProperty {
+    private var value: String = ImpressionsMode.optimized.rawValue
+    public var projectedValue: ImpressionsMode = .optimized
+    public var wrappedValue: String {
+        get {
+            return value
+        }
+
+        set {
+            setValue(newValue)
+        }
+    }
+
+    public init(wrappedValue: String) {
+        setValue(wrappedValue)
+    }
+
+    private mutating func setValue(_ newValue: String) {
+        let uppercased = newValue.uppercased()
+        guard let projectedValue = ImpressionsMode(rawValue: uppercased) else {
+                Logger.w("You passed an invalid impressionsMode (\(uppercased)), " +
+                    " impressionsMode should be one of the following values: " +
+                            "'DEBUG', 'OPTIMIZED' or 'NONE'. Defaulting to 'OPTMIZED' mode.")
```

Comment:
```suggestion
                            "'DEBUG', 'OPTIMIZED' or 'NONE'. Defaulting to 'OPTIMIZED' mode.")
```

Reply:
Thanks @gthea! will change that in next PR